src/main/java/sk/tomsik68/mclauncher/impl/login/yggdrasil/YDLoginService.java

```diff
@@ -22,12 +22,34 @@
 
 public final class YDLoginService implements ILoginService {
     public static UUID clientToken = UUID.randomUUID();
-    private static final String PASSWORD_LOGIN_URL = "https://authserver.mojang.com/authenticate";
-    private static final String SESSION_LOGIN_URL = "https://authserver.mojang.com/refresh";
-    private static final String SESSION_LOGOUT_URL = "https://authserver.mojang.com/invalidate";
+    private static String PASSWORD_LOGIN_URL = "https://authserver.mojang.com/authenticate";
+    private static String SESSION_LOGIN_URL = "https://authserver.mojang.com/refresh";
+    private static String SESSION_LOGOUT_URL = "https://authserver.mojang.com/invalidate";
 
     public YDLoginService() {
     }
+    
+    /**
+     * Constructor for debug/custom auth url.
+     * If argument is null, use default value
+     *
+     * @param passwordLoginUrl url for pass login and password
+     * @param sessionLoginUrl url for login by session (or refresh)
+     * @param sessionLogoutUrl url for logout by session
+     */
+    public YDLoginService(@Nullable String passwordLoginUrl,
```

Comment:
Do we really need to pass three urls? I don't think you ever need to authenticate against one server and then ask another (unrelated) server for a session token. I believe this could work with one URL, too. You would ask for a string in the constructor, for example `"https://authserver.mojang.com/"` and you can easily get the urls like `String passwordLoginUrl = authserverUrl + "authenticate"`.

Reply:
Good point! Thank you!

Reply:
What if we need other endpoint for `authenticate`/`refresh`/`invalidate`?

Reply:
I think it's sufficient that `YDLoginService` is compatible with mojang authentication server at the default url or a different one. If someone needs another way to obtain an `ISession`, they can just implement their own `ILoginService`.